src/ExtCmd.actor.cpp

```diff
@@ -90,9 +90,10 @@ ACTOR static Future<std::pair<int, int>> dropIndexMatching(Reference<DocTransact
 
 	matchingIndex->clearDescendants();
 	matchingIndex->clearRoot();
+
 	Void _ = wait(matchingIndex->commitChanges());
 
-	KeyRef indexKey = targetedCollection->getIndexesSubspace().toString() + encodeMaybeDotted(matchingName);
+	Key indexKey = targetedCollection->getIndexesSubspace().withSuffix(StringRef(encodeMaybeDotted(matchingName)));
```

Comment:
Can you help me understand what exactly is happening here? @apkar

Reply:
In that line, operator `+` is concatenating two `std::string` objects and creating a new `std::string` which is being referenced from `StringRef`. But that string itself gets destroyed after that line. `indexKey` is pointing to the dead memory in `std::string`.

Reply:
StringRef lacks the constructor which takes a rvalue then move the memory ownership, and thus once this line goes out of cope, memory pointed by the KeyRef becomes invalid. The fix changed this to use Standalone<StringRef> which manage its own memory on the arena.